### PR TITLE
Readiness

### PR DESCRIPTION
**Pull request by kvosper**
### :pencil: Description
- Adding a handler to the admin interface that indicates when Styx has completed started up

### :link: Related Issues
-

### CODE REVIEW

File: components/proxy/src/main/java/com/hotels/styx/admin/AdminServerBuilder.java [hunk 1]
```diff
@@ -226,6 +231,7 @@ private static Iterable<IndexHandler.Link> indexLinkPaths(StyxConfig styxConfig)
                 link("version.txt", "/version.txt"),
                 link("uptime", "/admin/uptime"),
                 link("Ping", "/admin/ping"),
+                link("Readiness", "/admin/readiness"),
```
Comment (NersesAM) [inline components/proxy/src/main/java/com/hotels/styx/admin/AdminServerBuilder.java:234]:
```suggestion
                link("Readiness", "/admin/ready"),
```

File: components/proxy/src/test/kotlin/com/hotels/styx/admin/handlers/ReadinessHandlerTest.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import com.hotels.styx.requestContext
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.StringSpec
+import reactor.core.publisher.toMono
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.text.Charsets.UTF_8
+
+class ReadinessHandlerTest : StringSpec() {
+    init {
+        "indicates readiness" {
```
Comment (NersesAM) [inline components/proxy/src/test/kotlin/com/hotels/styx/admin/handlers/ReadinessHandlerTest.kt:32]: this test is testing 2 different scenarios. Please split them into 2 separate tests.

Comment (kvosper) [reply components/proxy/src/test/kotlin/com/hotels/styx/admin/handlers/ReadinessHandlerTest.kt:32]: It is testing the scenario that the ReadinessHandler changes its content when the readiness changes, but I can also test the two states individually. I'll add those tests.

Comment (NersesAM) [reply components/proxy/src/test/kotlin/com/hotels/styx/admin/handlers/ReadinessHandlerTest.kt:32]: yeah you are right

**Review by NersesAM — APPROVED** [review]
LGTM

**Review by ryanellison — APPROVED** [review]

**Review by ryanellison — APPROVED** [review]

**Pull request merged**